lib/lib.error.php

```diff
@@ -45,93 +45,102 @@
+        ": ";
 
-	if ( $msglevel <= $verbose  ) {
-		// if this is the webinterface, print it to stdout, else to stderr
-		if ( IS_WEB ) {
-			$msg = htmlspecialchars($string);
-			// if this is the API, do not add HTML formatting and send HTTP status code
-			if ( defined('DOMJUDGE_API_VERSION') ) {
-				if ( $msglevel == LOG_ERR && ! headers_sent() ) {
-					$protocol = (isset($_SERVER['SERVER_PROTOCOL']) ?
-						$_SERVER['SERVER_PROTOCOL'] : 'HTTP/1.0');
-					header($protocol . " 500 Internal Server Error");
-				}
-				// Note: should we skip non-fatal messages in the API? A warning
-				// will probably mess up json output.
-				echo $msg . "\n";
-			// normal interactive web interface: output with markup
-			} else {
-				// string 'ERROR' parsed by submit client, don't modify!
-				if ( $msglevel == LOG_ERR ) {
-					echo "<fieldset class=\"error\"><legend>ERROR</legend> " .
-						 $msg . "</fieldset><!-- trigger HTML validator error: --><b>\n";
-				} else
-				if ( $msglevel == LOG_WARNING ) {
-					echo "<fieldset class=\"warning\"><legend>Warning</legend> " .
-						$msg . "</fieldset><!-- trigger HTML validator error: --><b>\n";
-				} else {
-					echo "<p>" . $msg . "</p>\n";
-				}
-			}
-		} else {
-			fwrite(STDERR, $stamp . $string . "\n");
-			fflush(STDERR);
-		}
-	}
-	if ( $msglevel <= $loglevel ) {
-		if ( defined('STDLOG') ) {
-			fwrite(STDLOG, $stamp . $string . "\n");
-			fflush(STDLOG);
-		}
-		if ( defined('SYSLOG') ) {
-			syslog($msglevel, $string . "\n");
-		}
-	}
+    if ($msglevel <= $verbose) {
+        // if this is the webinterface, print it to stdout, else to stderr
+        if (IS_WEB) {
+            $msg = htmlspecialchars($string);
+            // if this is the API, do not add HTML formatting and send HTTP status code
+            if (defined('DOMJUDGE_API_VERSION')) {
+                if ($msglevel == LOG_ERR && ! headers_sent()) {
+                    $protocol = (isset($_SERVER['SERVER_PROTOCOL']) ?
+                        $_SERVER['SERVER_PROTOCOL'] : 'HTTP/1.0');
+                    header($protocol . " 500 Internal Server Error");
+                }
+                // Note: should we skip non-fatal messages in the API? A warning
+                // will probably mess up json output.
+                echo $msg . "\n";
+            // normal interactive web interface: output with markup
+            } else {
+                // string 'ERROR' parsed by submit client, don't modify!
+                if ($msglevel == LOG_ERR) {
+                    echo "<fieldset class=\"error\"><legend>ERROR</legend> " .
+                         $msg . "</fieldset><!-- trigger HTML validator error: --><b>\n";
+                } elseif ($msglevel == LOG_WARNING) {
+                    echo "<fieldset class=\"warning\"><legend>Warning</legend> " .
+                        $msg . "</fieldset><!-- trigger HTML validator error: --><b>\n";
+                } else {
```

Comment:
How much is enforced by PSR-2? For example I see that the line continuations of the echo commands do not match up.

Reply:
I think this specific case is "left open" ("Inter-line alignment"), but there are other cases where they explicitly say that things must be indented "once" (i.e. with 4 spaces extra), for example when having function arguments on multiple lines.

I'd say that it would make sense to then also use this same rule for these multi-line echo statements, as this feels more consistent. But if others think we should match their alignment I won't fight it too hard.

Reply:
I'd prefer to leave it open. If we wanted to strictly enforce the same policy as for multiline function arguments, then we should write (with implicit parentheses for the `echo` function):
```
echo
    "<fieldset class=\"error\"><legend>ERROR</legend> " .
    $msg . "</fieldset><!-- trigger HTML validator error: --><b>\n";
```
which I personally dislike a lot. So I prefer to keep the flexibility to format this as the person coding sees fit.